frontend/packages/dev-console/src/components/pipelineruns/logs/log-snippet-utils.ts

```diff
@@ -0,0 +1,25 @@
+import { Condition } from '../../../utils/pipeline-augment';
+import { CombinedErrorDetails, LogSnippetTask } from './log-snippet-types';
+
+const joinConditions = (conditions: Condition[]) =>
+  conditions.map((condition) => condition.message).join('\n') || 'Unknown failure condition';
+
+export const taskRunSnippetMessage = (
+  taskRun: LogSnippetTask,
+  containerName: string,
+): CombinedErrorDetails => {
+  const taskName = taskRun.pipelineTaskName || taskRun.metadata.name;
```

Comment:
Hmm, these resources are so alike, but just different enough to dirty the implementation here.

I wonder if instead of passing a `LogSnippetTask` we could look to pass the status object and the name separately, so this internal method is unaware of the relationship between these two separate but same data structures.

Ie:
```suggestion
export const taskRunSnippetMessage = (
  taskName: string,
   // we may want to keep `taskRunStatus` as a stand alone type, but I expanded inline to show what I am referring to
  taskRunStatus: { podName: string, conditions: ConditionType[] },
  containerName: string,
): CombinedErrorDetails => {
  const taskName = taskRun.pipelineTaskName || taskRun.metadata.name;
```

Reply:
That was the case in the very beginning of the PR. And there would have been no need to keep a separate util for rendering out of  strings.

Reply:
updated